### PR TITLE
Add Config module using pydantic

### LINKED ISSUE #34: Config file + default mix

### PR DESCRIPTION
**Pull request by SarahAlidoost**
closes #34

### CODE REVIEW

File: dgl_ptm/dgl_ptm/config.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    model_config = ConfigDict(validate_default = True)
+
+
+class Config(BaseModel):
+    """Base class for configuration parameters."""
+    model_identifier: str = Field("test", alias='_model_identifier') # because pydantic does not like underscores
+    number_agents: PositiveInt = 100
+    initial_graph_type: str = "barabasi-albert"
+    model_data: dict = {}
+    model_graph: object = None # TODO confirm that model_graph is a class obj
```
Comment (meiertgrootes) [inline dgl_ptm/dgl_ptm/config.py:196]: confirmed that model graph is a class object

File: dgl_ptm/dgl_ptm/config.yaml [hunk 2]
```diff
@@ -21,7 +21,7 @@ steering_parameters:
   characteristic_distance: 35
```
Comment (meiertgrootes) [inline dgl_ptm/dgl_ptm/config.yaml:21]: We now have duplication in the parameters, as weight_a and weight_b are the deprecated names for `homophily_parameter` and `characteristic_distance`

File: dgl_ptm/dgl_ptm/config.py
Comment (meiertgrootes) [inline dgl_ptm/dgl_ptm/config.py:1]: This approach is very beneficial as it allows for input verification. The price, however, is the hardcoding of default parameters and config stuctures in the config.py code. In my opinion this is a trade-off worth making, but as adaptation of the model (and associated config) will require modification  of this file, we should maybe add more elaborate inline documentation of the function and purpose of the individual elements, as well as maybe bundle functions more. This need not be blocking for merging, but at least warrants a future issue

**Review by meiertgrootes — APPROVED** [review]
Use of pydantic for config appears beneficial. Some comments on clarity for later use. Can also be addressed in new issue after merging.

File: dgl_ptm/dgl_ptm/config.yaml (hunk 2 above)
Comment (SarahAlidoost) [reply dgl_ptm/dgl_ptm/config.yaml:21]: They are removed.

File: dgl_ptm/dgl_ptm/config.py
Comment (SarahAlidoost) [reply dgl_ptm/dgl_ptm/config.py:1]: issue #48 submitted.

**Pull request merged**